erts/etc/win32/wsl_tools/vc/cc.sh

```diff
@@ -29,6 +29,14 @@ SAVE="$@"
 # Constants
 COMMON_CFLAGS="-nologo -D__WIN32__ -DWIN32 -DWINDOWS -D_WIN32 -DNT -D_CRT_SECURE_NO_DEPRECATE"
 
+if [ "$CONFIG_SUBTYPE" = "x64_arm64" -o "$CONFIG_SUBTYPE" = "arm64" ]; then
+	MACHINE="ARM64"
+	STDCXX_VERSION="/std:c++20"
```

Comment:
I think there're some one-offs in the code base, or at least LLVM think they should be in C++20 standard. For example,

```
CXX    obj/win32/opt/jit/beam_asm_global.o
beam\jit\arm\beam_asm_global.cpp(77): error C7555: use of designated initializers requires at least '/std:c++20'
Failed: cl.exe -nologo -D__WIN32__ -DWIN32 -DWINDOWS -D_WIN32 -DNT -D_CRT_SECURE_NO_DEPRECATE -D__aarch64__ -MD -Ox -Z7 /std:c++17 /Zc:__cplusplus -DASMJIT_EMBED=1 -DASMJIT_NO_BUILDER=1 -DASMJIT_NO_DEPRECATED=1 -DASMJIT_STATIC=1 -DASMJIT_NO_FOREIGN=1 -Iwin32/opt/jit -Ibeam -Isys/win32 -Isys/common -Iwin32 -Izlib -Ipcre -Iryu -Iopenssl/include -I../include -I../include/win32 -I../include/internal -I../include/internal/win32 -Ibeam/jit -Ibeam/jit/arm -I"C:\src\erts\win32" -D_WIN32_WINNT=0x0600 -DWINVER=0x0600 -DERTS_MIXED_VC -DSTATIC_ERLANG_DRIVER -DHAVE_CONFIG_H -DUSE_THREADS -DWIN32_THREADS -DBEAMASM=1 /std:c++17 /Zc:__cplusplus /EHsc /FS -c -FoC:\src\erts\emulator\obj\win32\opt\jit\beam_asm_global.o beam\jit\arm\beam_asm_global.cpp
make[4]: *** [win32/Makefile:947: obj/win32/opt/jit/beam_asm_global.o] Error 2
make[4]: Leaving directory '/mnt/c/src/erts/emulator'
make[3]: *** [/mnt/c/src/make/run_make.mk:35: opt] Error 2
make[3]: Leaving directory '/mnt/c/src/erts/emulator'
make[2]: *** [Makefile:45: opt] Error 2
make[2]: Leaving directory '/mnt/c/src/erts'
make[1]: *** [Makefile:60: jit] Error 2
make[1]: Leaving directory '/mnt/c/src/erts'
make: *** [Makefile:483: emulator] Error 2
```
 I'll try to fix these one-offs and keep the minimum version to be C++17.

Reply:
### update
Designated initializers seem to be indeed available since C++20, [reference link](https://en.cppreference.com/w/cpp/language/aggregate_initialization).

Reply:
Fixed in [7e9801f](https://github.com/erlang/otp/pull/8142/commits/7e9801fdb230e236033f89128e990b5925d0f03d)